scripts/create-kubestellar-demo-env.sh

```diff
@@ -17,6 +17,38 @@
 
 set -e
 
+echo -e "Starting Docker engine..."
+# Check if Docker is running using OS-agnostic commands
+if ! docker info >/dev/null 2>&1; then
+    case "$(uname -s)" in
+        Linux*)
+            sudo systemctl start docker
+            ;;
+        Darwin*)
+            open -a Docker
```

Comment:
On my Mac, where I am using Rancher Desktop and the docker container runtime:

```console
mspreitz@mjs13 kubestellar % open -a Docker
Unable to find application named 'Docker'
```

In general, remember that there is commonly a "desktop" application that provides a UI and the thing that actually provides the container runtime is something else.